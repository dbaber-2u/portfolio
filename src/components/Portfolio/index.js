

const projects = [
    { name: "Project 1", root: "project1" },
    { name: "Project 2", root: "project2" },
    { name: "Project 3", root: "project3" },
    { name: "Project 4", root: "project4" },
    { name: "Project 5", root: "project5" },
    { name: "Project 6", root: "project6" }
];

function Portfolio() {
    return (
        <section>
            <div className="w3-panel w3-red">
                <p>The following are examples and not real</p>
            </div>
            <div className="flex-row">
                {projects.map((prj, i) => (
                <div>
                <p><a href={`https://github.com/dbaber-2u/${prj.root}`}>{prj.name}</a></p>
                <img
                    src={require(`../../assets/programming${i+1}.jpg`)}
                    className="img-thumbnail mx-1"
                    alt={prj.name}
                    href={`http://dbaber-2u.github.io/${prj.root}`}
                />
                </div>    
                ))}
            </div>
            
        </section>
    )
}

export default Portfolio;