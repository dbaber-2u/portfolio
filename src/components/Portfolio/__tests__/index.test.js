import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '..';

afterEach(cleanup);

describe('Navigation component', () => {
    // baseline test
    it('renders', () => {
        render(<Portfolio />);
    });

    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Portfolio />);

        expect(asFragment()).toMatchSnapshot();
    });
})