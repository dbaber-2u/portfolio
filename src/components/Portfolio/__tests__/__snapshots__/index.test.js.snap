// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="w3-panel w3-red"
    >
      <p>
        The following are examples and not real
      </p>
    </div>
    <div
      class="flex-row"
    >
      <div>
        <p>
          <a
            href="https://github.com/dbaber-2u/project1"
          >
            Project 1
          </a>
        </p>
        <img
          alt="Project 1"
          class="img-thumbnail mx-1"
          href="http://dbaber-2u.github.io/project1"
          src="programming1.jpg"
        />
      </div>
      <div>
        <p>
          <a
            href="https://github.com/dbaber-2u/project2"
          >
            Project 2
          </a>
        </p>
        <img
          alt="Project 2"
          class="img-thumbnail mx-1"
          href="http://dbaber-2u.github.io/project2"
          src="programming2.jpg"
        />
      </div>
      <div>
        <p>
          <a
            href="https://github.com/dbaber-2u/project3"
          >
            Project 3
          </a>
        </p>
        <img
          alt="Project 3"
          class="img-thumbnail mx-1"
          href="http://dbaber-2u.github.io/project3"
          src="programming3.jpg"
        />
      </div>
      <div>
        <p>
          <a
            href="https://github.com/dbaber-2u/project4"
          >
            Project 4
          </a>
        </p>
        <img
          alt="Project 4"
          class="img-thumbnail mx-1"
          href="http://dbaber-2u.github.io/project4"
          src="programming4.jpg"
        />
      </div>
      <div>
        <p>
          <a
            href="https://github.com/dbaber-2u/project5"
          >
            Project 5
          </a>
        </p>
        <img
          alt="Project 5"
          class="img-thumbnail mx-1"
          href="http://dbaber-2u.github.io/project5"
          src="programming5.jpg"
        />
      </div>
      <div>
        <p>
          <a
            href="https://github.com/dbaber-2u/project6"
          >
            Project 6
          </a>
        </p>
        <img
          alt="Project 6"
          class="img-thumbnail mx-1"
          href="http://dbaber-2u.github.io/project6"
          src="programming6.jpg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
